contrib/xor-blocks/src/main.rs

```diff
@@ -0,0 +1,193 @@
+// Copyright (c) The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+use std::{
+    env,
+    fmt::{self, Debug, Display, Formatter},
+    fs::{self, File},
+    hash::{BuildHasher, Hasher, RandomState},
+    io::{self, BufReader, BufWriter, Read, Write},
+    path::Path,
+    time::{Duration, Instant},
+};
+
+const MAGIC: [u8; 4] = [0xf9, 0xbe, 0xb4, 0xd9];
+const XOR_FILE_NAME: &str = "xor.dat";
+
+#[derive(Debug)]
```

Comment:
I see you're using this to display the messages from `main`. However that will lead to horrible messages containing `Error { msg: "..." }`. I suggest you implement `Debug` manually instead just displaying the string.